test/test_linalg.py

```diff
@@ -921,6 +922,117 @@ def test_nuclear_norm_exceptions_old(self, device):
         self.assertRaisesRegex(RuntimeError, "duplicate or invalid", torch.norm, x, "nuc", (0, 0))
         self.assertRaisesRegex(IndexError, "Dimension out of range", torch.norm, x, "nuc", (0, 2))
 
+    @skipCUDAIfNoMagmaAndNoCusolver
+    @skipCPUIfNoLapack
+    @dtypes(torch.float32, torch.float64, torch.complex64, torch.complex128)
+    @precisionOverride({torch.float32: 2e-3, torch.complex64: 2e-3})
+    def test_inverse(self, device, dtype):
+        from torch.testing._internal.common_utils import random_fullrank_matrix_distinct_singular_value
+
+        def run_test(matrix, batches, n):
+            matrix_inverse = torch.inverse(matrix)
+
+            # Compare against NumPy output
+            # NumPy uses 'gesv' LAPACK routine solving the equation A A_inv = I
+            # But in PyTorch 'gertf' + 'getri' is used causing element-wise differences
+            expected = np.linalg.inv(matrix.cpu().numpy())
+            self.assertEqual(matrix_inverse, expected, atol=self.precision, rtol=1e-4)
+
+            # Additional correctness tests, check matrix*matrix_inverse == identity
+            identity = torch.eye(n, dtype=dtype, device=device)
+            # TODO(@ivanyashchuk): remove this once batched matmul is avaiable on CUDA for complex dtypes
+            if 'cuda' in device and dtype.is_complex:
```

Comment:
Don't check for the string to determine the device type. Use either `torch.device(device).device_type == 'cuda'` or `self.device_type == 'cuda'`.